src/sentry/api/endpoints/organization_member/details.py

```diff
@@ -508,15 +508,16 @@ def delete(
             )
 
         with transaction.atomic(router.db_for_write(OrganizationMember)):
-            # Delete any invite requests and pending invites by the deleted member
-            existing_invites = OrganizationMember.objects.filter(
-                Q(invite_status=InviteStatus.REQUESTED_TO_BE_INVITED.value)
-                | Q(token__isnull=False),
-                inviter_id=member.user_id,
-                organization=organization,
-            )
-            for om in existing_invites:
-                om.delete()
+            if member.user_id:
+                # Delete any invite requests and pending invites by the deleted member
+                existing_invites = OrganizationMember.objects.filter(
+                    Q(invite_status=InviteStatus.REQUESTED_TO_BE_INVITED.value)
+                    | Q(token__isnull=False),
```

Comment:
If our invite accept logic does not remove the token then this line should be deleted really. Will confirm tomorrow.

Reply:
@oioki can you confirm if it's safe to remove this line? Basically what happens if we only look at InviteStatus and don't rely on token at all?

Reply:
Feels like it may warrant the extra effort of adding a test covering this specific case, btw.

Reply:
I'm going to leave the token line as is because I'm not 100% sure that it'll break logic that Alex put it before. It also doesn't affect the fix because this line doesn't get called anyways after I added the `if`.